src/electionguard/decryption_share.py

```diff
@@ -105,46 +245,67 @@ class CompensatedDecryptionShare:
     an available guardian on behalf of a missing guardian
     """
 
-    # The Available Guardian that this partial share belongs to
-    available_guardian_id: GUARDIAN_ID
+    guardian_id: GUARDIAN_ID
+    """
+    The Available Guardian that this share belongs to
+    """
 
-    # The Missing Guardian for whom this share is calculated on behalf of
     missing_guardian_id: GUARDIAN_ID
+    """
+    The Missing Guardian for whom this share is calculated on behalf of
+    """
+
+    public_key: ElementModP
+    """
+    The election public key for the guardian
+    """
 
-    # The collection of all contests in the election
     contests: Dict[CONTEST_ID, CiphertextCompensatedDecryptionContest]
+    """
+    The collection of decryption shares for all contests in the election
+    """
 
-    lagrange_coefficient: ElementModQ
+    spoiled_ballots: Dict[BALLOT_ID, CompensatedBallotDecryptionShare]
+    """
+    The collection of decryption shares for all spoiled ballots in the election
+    """
 
 
-def get_cast_shares_for_selection(
+def get_tally_shares_for_selection(
     selection_id: str, shares: Dict[GUARDIAN_ID, DecryptionShare],
-) -> Dict[GUARDIAN_ID, ElementModP]:
+) -> Dict[GUARDIAN_ID, Tuple[ElementModP, CiphertextDecryptionSelection]]:
```

Comment:
Instead of ElementModP <--- This should be named. I think this would improve readability

`PublicKeyShare = ElementModP` etc.